fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/provider/impl/AzureProvider.java

```diff
@@ -0,0 +1,387 @@
+     * @param container the container (also known as a bucket)
+     * @param chunkData the export data
+     * @param cosBucketPathPrefix path prefix
+     * @param fhirResourceType resource type
+     * @param currentBytes the long value or location of the end of the last resource
+     */
+    protected void registerOverride(String connectionString, String container, ExportTransientUserData chunkData,
+        String cosBucketPathPrefix, String fhirResourceType, long currentBytes) {
+        this.connectionString = connectionString;
+        this.container = container;
+        this.chunkData = chunkData;
+        this.cosBucketPathPrefix = cosBucketPathPrefix;
+        this.fhirResourceType = fhirResourceType;
+        this.currentBytes = currentBytes;
+    }
+
+    /**
+     * initializes the blob client
+     * @param workItem
+     */
+    protected void initializeBlobClient(String workItem) {
+        if (client == null) {
+            client = new BlobClientBuilder()
+                .connectionString(connectionString)
+                .containerName(container)
+                .blobName(workItem)
+                .buildClient();
+        }
+    }
+
+    @Override
+    public void createSource() throws FHIRException {
+        try {
+            BlobContainerClient blobContainerClient = new BlobContainerClientBuilder()
+                .connectionString(connectionString)
+                .containerName(container)
+                .buildClient();
+            blobContainerClient.create();
+            LOG.info("Container is created '" + container + "'");
+        } catch (BlobStorageException error) {
+            LOG.fine("Can't create container. It already exists");
+            LOG.throwing(this.getClass().getName(), "createSource", error);
+        }
+    }
+
+    @Override
+    public long getSize(String workItem) throws FHIRException {
+        try {
+            initializeBlobClient(workItem);
+            long size = client.getProperties().getBlobSize();
+            LOG.fine(() -> workItem + " [" + size + "]");
+            return size;
+        } catch (Exception e) {
+            LOG.throwing("AzureProvider", "getSize", e);
+            LOG.fine("Error Getting File Size '" + container + "/" + workItem + "'");
+            throw new FHIRLoadException("Error Getting File Size '" + container + "/" + workItem + "'");
+        }
+    }
+
+    /**
+     * lists the blobs on the container.
+     * @throws FHIRException
+     */
+    public void listBlobsForContainer() throws FHIRException {
+        try {
+            BlobContainerClient containerClient = new BlobContainerClientBuilder()
+                    .connectionString(connectionString)
+                    .containerName(container)
+                    .buildClient();
+            LOG.info("Listing the blobs");
+            containerClient.listBlobs().forEach(blob -> LOG.info("BLOB: " + blob.getName()));
+            LOG.info("Finished listing the blobs");
```

Comment:
does this get called much?  if so, we should consider reducing the log level of these to fine.  but the only reference I found was from a test, so maybe its OK.

Reply:
it's really designed for test situations/debug with azure blob.